### PR TITLE
Makefile targets for cross build and distributions

### PR DESCRIPTION
**Pull request by dims**
example:
`VERSION=0.1.0 make build-cross dist`

The command line above should build a bunch of tars and zips that can
be uploaded when we cut a release.

Code was borrowed from https://github.com/kubernetes/helm/blob/master/Makefile

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -231,5 +237,29 @@ upload-images: images
 version:
 	@echo ${VERSION}
 
+.PHONY: build-cross
+build-cross: LDFLAGS += -extldflags "-static"
+build-cross:
+ifndef HAS_GOX
+	go get -u github.com/mitchellh/gox
+endif
+	CGO_ENABLED=0 gox -parallel=3 -output="_dist/{{.OS}}-{{.Arch}}/{{.Dir}}" -osarch='$(TARGETS)' $(GOFLAGS) $(if $(TAGS),-tags '$(TAGS)',) -ldflags '$(LDFLAGS)' $(GIT_HOST)/$(BASE_DIR)/cmd/openstack-cloud-controller-manager/
```
Comment (j-griffith) [inline Makefile]: Any reason to not make the number of parallel compiles configurable?

File: Makefile [hunk 2]
```diff
@@ -231,5 +237,29 @@ upload-images: images
 version:
 	@echo ${VERSION}
 
+.PHONY: build-cross
+build-cross: LDFLAGS += -extldflags "-static"
+build-cross:
```
Comment (j-griffith) [inline Makefile]: Doesn't need to be a part of this change, but I'm wondering if we should consider having "make dep" be required task for some of these, or at least a check if vendor exists and just do it if not.  Again, not an issue for this change but this is a good opportunity to capture the idea.

File: Makefile (hunk 1 above)
Comment (j-griffith) [inline Makefile]: output to _dist makes me SOOO HAPPY!

File: Makefile [hunk 3]
```diff
@@ -231,5 +237,29 @@ upload-images: images
...
+.PHONY: dist
+dist:
+	( \
+		cd _dist && \
+		$(DIST_DIRS) cp ../LICENSE {} \; && \
+		$(DIST_DIRS) cp ../README.md {} \; && \
+		$(DIST_DIRS) tar -zcf cloud-provider-openstack-$(VERSION)-{}.tar.gz {} \; && \
+		$(DIST_DIRS) zip -r cloud-provider-openstack-$(VERSION)-{}.zip {} \; \
+	)
+
```
Comment (j-griffith) [inline Makefile]: This is great, but we should consider cleaning up all the artifacts.  The way it stands now we end up with 3 items for each dist:
```
<dist>.tar.gz
<dist>.zip
<dist>/
```

I assume by default most that run this are going to be interested in just the tgz, or the zip.  The other option providing the most flexibility I guess is just provide the binaries (no tar, no zip) and then add a "make bundle" option to bundle them up

File: Makefile (hunk 2 above)
Comment (dims) [reply Makefile]: k added dependencies to the build-cross and dist target

File: Makefile (hunk 1 above)
Comment (dims) [reply Makefile]: Done

File: Makefile (hunk 3 above)
Comment (dims) [reply Makefile]: ack, please help tweak it. i just wanted us to get going

Comment (flaper87) [reply Makefile]: Great suggestion. I'll push this through so we can iterate over what's here.

**Comment by flaper87** [comment]
The error is not related to this change so I'm sending it through.

/lgtm

**Pull request merged**